DataCapturing/Persistence/PersistenceLayer.swift

```diff
@@ -315,6 +318,7 @@ public class PersistenceLayer {
         let fetchRequest: NSFetchRequest<MeasurementMO> = MeasurementMO.fetchRequest()
         // The following needs to use an Objective-C number. That is why `measurementIdentifier` is wrapped in `NSNumber`
         fetchRequest.predicate = NSPredicate(format: "identifier==%@", NSNumber(value: identifier))
+        //fetchRequest.relationshipKeyPathsForPrefetching = ["tracks.locations"]
```

Comment:
This would improve performance, since it loads the complete measurement at once into main memory. But it seems to me, this is already happening, so I left this commented out.